plugins/modules/aci_fabric_node.py

```diff
@@ -243,10 +260,19 @@ def main():
     switch = module.params.get("switch")
     description = module.params.get("description")
     role = module.params.get("role")
+    node_type = module.params.get("node_type")
+    remote_leaf_pool_id = module.params.get("remote_leaf_pool_id")
     state = module.params.get("state")
     name_alias = module.params.get("name_alias")
 
     aci = ACIModule(module)
+
+    if node_type != "remote" and remote_leaf_pool_id is not None:
+        module.fail_json(msg="External Pool Id is not compatible with a node type other than 'remote'.")
+    else:
+        if node_type is not None:
+            node_type = NODE_TYPE_MAPPING.get(node_type)
```

Comment:
I think the else statement is not needed. 

Also the second if conditional is not needed because .get() return None as default when key not found.

And I think that the if is not None is not needed and simply a value would resolve in True statement.

```suggestion
    if node_type != "remote" and remote_leaf_pool_id:
        module.fail_json(msg="External Pool Id is not compatible with a node type other than 'remote'.")
    
    node_type = NODE_TYPE_MAPPING.get(node_type)
```